cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/BodyConsumerAdapter.java

```diff
@@ -133,9 +127,15 @@ public void setTransactionFailureResponse(Throwable t) {
       }
 
       @Override
-      public void execute() {
+      public void execute(boolean keepAlive) {
         // no-op
       }
+
+      @Override
+      public boolean hasContentProducer() {
+        // Always release the context at the end since it's not possible to send with a content producer
+        return false;
```

Comment:
For this case, when the `handleError` method gets called from netty-http, a response has already sent (by the library), so it doesn't really matter what the `doSend` method get (and in fact, send->doSend only gets call from the `onError` method, which is a no-op anyway).